addons/website_profile/static/src/components/profile_dialog/profile_dialog.js

```diff
@@ -0,0 +1,149 @@
+import { Wysiwyg } from "@html_editor/wysiwyg";
+import { onMounted, onWillStart, reactive, useRef, useState } from "@odoo/owl";
+import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { localization } from "@web/core/l10n/localization";
+import { _t } from "@web/core/l10n/translation";
+import { rpc } from "@web/core/network/rpc";
+import { user } from "@web/core/user";
+import { useAutofocus, useService } from "@web/core/utils/hooks";
+import { isHtmlEmpty } from "@web/core/utils/html";
+import { isEmail, uuid } from "@web/core/utils/strings";
+import { setCursorEnd } from "@web_editor/js/editor/odoo-editor/src/utils/utils";
+
+export class ProfileDialog extends ConfirmationDialog {
+    static template = "website_profile.ProfileDialog";
+    static components = {
+        ...ConfirmationDialog.components,
+        Wysiwyg,
+    };
+    static props = {
+        ...ConfirmationDialog.props,
+        focusWebsiteDescription: {
+            type: Boolean,
+            optional: true,
+        },
+        userId: { type: Number },
+    };
+    static defaultProps = {
+        ...ConfirmationDialog.defaultProps,
+        cancel: () => {},
+        cancelLabel: _t("Discard"),
+        confirm: () => true,
+        confirmLabel: _t("Update"),
+        dismiss: () => {},
+        focusWebsiteDescription: false,
+        title: _t("Edit Profile"),
+    };
+
+    setup() {
+        super.setup();
+        this.orm = useService("orm");
+        const componentUUID = uuid();
+        this.getUniqueDomID = (suffix) => `profile-dialog-${componentUUID}-${suffix}`;
+        this.upload = useRef("upload");
+        this.profileImg = useRef("profileImg");
+        this.profileImgData = null;
+        this.state = useState({ hasError: false, emailHasError: false, nameHasError: false });
+        useAutofocus({ refName: "name" });
+        const websiteDescriptionClass = this.getUniqueDomID("websiteDescription");
+        onWillStart(async () => {
```

Comment:
Detail: We usually space out things a bit, blank space before "onWillStart", blank space before "onMounted" etc, makes it less "packed".